xo-archiver

```diff
@@ -442,14 +487,89 @@ function copy_to_s3 {
     if aws s3 cp $WORKING_DIR/ $_remote_xva_path/ --recursive
     then
         log "Successfully copied exports of \"$_vm_name\" on bucket \"$AWS_BUCKET_XO_ARCHIVES\""
-        log "Cleaning up $WORKING_DIR/"
-        rm -Rf $WORKING_DIR/
+        clean_working_dir
     else
         die "Fail to push to s3"
     fi
 }
 
 
+# Usage: copy_from_s3 <VM-NAME>
+# copies the <VM-NAME> corresponding files from the s3 bucket specified through envvar
+function copy_from_s3 {
+    if [[ -z $1 ]]
+    then
+        die "copy_from_s3 needs parameter: VM-NAME. E.g: foo-bar-01"
+    fi
+
+    # ensure bucket var is set
+    if [[ -z "$AWS_BUCKET_XO_ARCHIVES" ]]
+    then 
+        die "Bucket is undefined. look at the --help and search for AWS_BUCKET_XO_ARCHIVES"
+    fi
+
+    create_working_dir
+
+    local _remote_xva_path="s3://$AWS_BUCKET_XO_ARCHIVES/$1"
+    log "Copying archives from \"$_remote_xva_path\" into $WORKING_DIR/"
+
+    if aws s3 cp $_remote_xva_path/ $WORKING_DIR/ --recursive --force-glacier-transfer
```

Comment:
From what I understand, you can be retrieving Glacier archives here. What kind of Glacier retrieval is performed when you copy the archive on your local filesystem? I suppose it's the expedited retrieval, else it would take forever :thinking:

Reply:
With Glacier, you need to `restore` a file before being able to Get it.
There are several ways to restore and each can take from several minutes to hours.

To keep this script as simple as possible, it doesn't proceed the Glacier `restore` operation. As such, there's no terraform/cloudformation code to create a bucket or handle lifecycle policies in this script.
The only AWS interaction this script has is to copy some files. I want to keep that simplicity.

Therefore, if there is some Glacier in the mix, the user of this script needs to ensure the aws command is able to copy a file and by such, she/he needs to `restore` the necessary objects as needed beforehand.